Upstream sync june6

JohnGarbutt opened the pull request.

Commit:
Allow custom fluentd input configurations

Co-authored-by: Mark Goddard <mark@stackhpc.com>
Change-Id: I2e5ecf5b01cc842ec480fc4d883a7d2283fc1c31

dougszumski committed:
Support deploying Monasca Log Metrics

The log metrics service generates metrics from log messages
which allows further analysis and alerting to be performed
on them. Basic configuration is provided so that metrics
are generated for high level warning logs such as error, or
warning.

Change-Id: I45cc17817c716296451f620f304c0b1108162a56
Partially-Implements: blueprint monasca-roles

Commit:
[prometheus] Enable ceph mgr exporter

This patch enables the ceph mgr prometheus exporter.

If enable_prometheus_ceph_mgr_exporter is set to true,
the ceph mgr prometheus plugin is enabled on the hosts that are part
of the ceph-mgr group, then the exporter is added into the prometheus-server
configuration file.

Change-Id: Ia2f879401e585e6043f69cc5e3ab1a1f72f7f033

Conflicts:
	ansible/group_vars/all.yml
	ansible/roles/prometheus/templates/prometheus.yml.j2
	etc/kolla/globals.yml

dougszumski committed:
Support deploying an Apache Storm cluster

Storm is required for running the Monasca thresholder component for
generating alerts.

Change-Id: I5e1ef74dc55a787293abbb3e629b5ab1ce5f4bbb
Partially-Implements: blueprint monasca-roles

dougszumski committed:
Support deploying Monasca Thresh

Monasca Thresh is a Storm topology which generates alerts from
metric streams according to alarms defined via the Monasca API.

This change runs the thresholder in local mode, which means that
the log output for the topology is directed to stdout and the
topology is restarted if the container is restarted. A future
change will improve the log collection and introduce a better
way of the checking the topology is running for multi-node
clusters.

Partially-Implements: blueprint monasca-roles
Change-Id: I063dca5eead15f3cec009df62f0fc5d857dd4bb0

dougszumski committed:
Support deploying Monasca Persister

The Monasca Persister reads metrics from Kafka and stores them
in a configurable time series database.

Change-Id: I8166b32bfb1583098ab8318a5f38d25bddb81e89
Partially-Implements: blueprint monasca-roles

jovial committed:
Create /volumes when using external ceph

Currently we only create the /volumes directory when
using the kolla-ansible bundled ceph. This extends the
functionality to external ceph.

Note: In addition to the cherry-picked commit, we add jmespath to
requirements as it is required for the json_query filter, with a minimum
version of 0.9.3 to match lower-constraints.txt.

Change-Id: I1e9a40e4ee893a53f04909ef7835be005c9b5276
Closes-Bug: #1778720
(cherry picked from commit 2f6b1c6890cf7ea6b0dd33ac219646e4dcaf1fd6)

dougszumski committed:
Support deploying Monasca Notification engine

The Monasca Notification engine generates alerts such as Slack
notifications from alerts.

Change-Id: I84861d5feefe6b6f38acc4dd71e94c386d40b562
Partially-Implements: blueprint monasca-roles

dougszumski committed:
Add region to Monasca Log API

This is required for upcoming log query support and it also
causes an error in the Keystone middleware if it's missing.

Partially-Implements: blueprint monasca-roles
Change-Id: I2bcb32bc0c079c799d2b0e45a97b454d38896986

dougszumski committed:
Support deploying the Monasca Agent

The Monasca Agent collects metrics and in this change is deployed
across the control plane. These metrics are collected into an OpenStack
project. It supports configuring a small number of plugins, which can
be extended in later commits. It also makes the Monasca Agent credentials
available to other roles, such as the common role to allow forwarding
logs to Monasca.

Partially-Implements: blueprint monasca-roles
Change-Id: I76b34fc5e1c76407a45fcf272268d5798b473ca2

dougszumski committed:
Improve scalability of Kafka

Set sensible defaults for replica counts and minimum insync replicas
as a function of the number of nodes in the Kafka cluster.

Partially-Implements: blueprint monasca-roles
Change-Id: Icf1dddb7dd6a64f4e5efb7dffa5ffdf0880f891f

dougszumski committed:
Improve registration of Monasca InfluxDB database

Monasca is not yet compatible with InfluxDB > 1.1.10, which means
that the official Ansible modules for managing InfluxDB don't work [1].
We therefore fall back to manual commands to register the database
and a default retention policy.

[1] https://github.com/influxdata/influxdb-python
    #influxdb-pre-v110-users

Partially-Implements: blueprint monasca-roles
Change-Id: I59ceda1e7a6e945b13872089011045db04548b94

dougszumski committed:
Use alembic migrations to bootstrap Monasca database

Alembic uses the monasca-agent config file for the DB
credentials. These are therefore no longer required.

Partially-Implements: blueprint monasca-roles
Change-Id: Id076e7a0b838888314159dd9e76697f89adecb5e

dougszumski committed:
Prevent Monasca Notification using YAML mode

If the config option is not specified Monasca Notification defaults
to YAML config parsing and fails to load.

Partially-Implements: blueprint monasca-roles
Change-Id: Id85d9765df525b34d43a77a9e4045e0de19aef99

dougszumski committed:
Forward system logs to Monasca Log API

When Monasca is enabled disable direct logging to ElasticSearch and
send all logs harvested by Fluentd to the Monasca Log API.

This change also cleans up output files which may be left behind when
the various log forwarding options are enabled / disabled.

Partially-Implements: blueprint monasca-roles
Change-Id: I7197966c5117176407d60c86c08d3bcea5e8131a

dougszumski committed:
Increase maximum payload size for Monasca Log API

Increase the maximum payload size so that it slightly exceeds the default
maximum chunk size (8MB) in Fluentd. This prevents the Monasca Log API
dropping logs when under heavy load.

Partially-Implements: blueprint monasca-roles
Change-Id: I13020abde7de1c92acc07352466b814b4a6ce5bc

dougszumski committed:
Update Monasca Log Persister config

* Make number of threads configurable
* Improve naming
* Update path and log options for Logstash 2.4
* Use the correct logs directory

Partially-Implements: blueprint monasca-roles
Change-Id: I09388de4d31740f2786393c957201bd037572226

dougszumski committed:
Update Monasca Log Transformer config

* Convert all log levels to lowercase
* Make number of threads configurable
* Improve naming
* Update path and log options for Logstash 2.4
* Change worker count from string to integer for the Kafka output plugin supplied
  with Logstash 2.4.

Partially-Implements: blueprint monasca-roles
Change-Id: I9379969ee639dff0b525d7a1a03df59bcf18b98b

dougszumski committed:
Remove Kafka partitions config option

This option doesn't actually do anything and a bug to remove
it from the Monasca API config file has been raised.

Partially-Implements: blueprint monasca-roles
Change-Id: I7ec1786b5828ab0135ca86ec040f83a6f4c78d9f

dougszumski committed:
Support configuring Monasca Persister performance

On a single node deployment, the Monasca persister can
limit the rate at which Monasca can persist metrics to
InfluxDB. Increasing the thread count can remove this
bottle neck.

Partially-Implements: blueprint monasca-roles
Change-Id: I763a5ae6aa8c8ab3bf766ab5b58c386da74a188b

dougszumski committed:
Support configuring Monasca log pipeline performance

Change-Id: Id8948fcf2d165f8285c7562e7aebd4145c4ff0db
Partially-Implements: blueprint monasca-roles

dougszumski committed:
Add some missing parameters for Monasca Notification

Partially-Implements: blueprint monasca-roles
Change-Id: I21de7748156f8d3689ebfc29f2fc4dc5f7f36ddf

dougszumski committed:
Add missing Monasca persister and notification dimensions

Partially-Implements: blueprint monasca-roles
Change-Id: I2fb43387bd1bc3b70f5c23d5268a7a7154ec31f0

dougszumski committed:
Add initial documentation for Monasca

Partially-Implements: blueprint monasca-roles
Change-Id: I19db9396bb53b6db77cf97d7969ca24a55d8db0e

dougszumski committed:
Support using binary Logstash image

To build the Monasca images, they need to be built from source, however,
this doesn't include Logstash. Logstash does not differentiate between
binary or source install types, so we can just use the default.

Partially-Implements: blueprint monasca-roles
Change-Id: I7eb355f25864f2b993780cd2b000c8bcd13f78e8

dougszumski committed:
Update log metrics config for Logstash downgrade

The format for the Logstash Kafka output plugin worker count is
integer in Logstash 2 and string in Logstash 5. Since we have downgraded
Logstash we need to switch formats here.

Change-Id: I98113dda05bbb44410916c27d7d3bcd59a034a98
Partially-Implements: blueprint monasca-roles

jeffrey4l committed:
Configure coordination in default for cinder when redis is enabled

Coordination is required for active-active cinder volume.

Change-Id: I9b26831fd951a24c483840a55824fae06ccbbf81
Closes-Bug: #1796615
(cherry picked from commit cf70bf23a5472027b4866bba5f17b6acb0e92b1b)

dougszumski committed:
Disable InfluxDB reporting

Make reporting of InfluxDB statistics opt-in rather than opt-out.

Trivial-Fix
Change-Id: I23342f797d883da1fe4c3ecdda53dd677239dc36

dougszumski committed:
Support custom monasca-notification templates

Jira, Slack and possiblly other plugins allow custom templates
for defining the format of notifications. This change lets you
provide these in a templates folder which is copied into the
monasca-notification container.

Partially-Implements: blueprint monasca-roles
Change-Id: Ibc5ba3944d51f6c8ffc8bdc9ed60f43dd91ca7e0

dougszumski committed:
Constrain the size of Docker logs

Even though Kolla services are configured to log output to file rather than
stdout, some stdout still occurs when for example the container re(starts).
Since the Docker logs are not constrained in size, they can fill up the
docker volumes drive and bring down the host. One example of when this is
particularly problematic is when Fluentd cannot parse a log message. The
warning output is written to the Docker log and in production we have seen
it eat 100GB of disk space in less than a day. We could configure Fluentd
not to do this, but the problem may still occur via another mechanism.

Change-Id: Ia6d3935263a5909c71750b34eb69e72e6e558b7a
Closes-Bug: #1794249
(cherry picked from commit bd54b99132d019b747dcd98af9371628625c8a1b)

dougszumski committed:
Monasca specific config for Fluentd

This commit adds some filters which format logs so that they
can be correctly sent to the Monasca Log API by the Monasca
Fluentd plugin. In the future the Fluentd plugin could be
extended and this config could be removed.

Partially-Implements: blueprint monasca-roles
Change-Id: I87b6dfb3052d03f87349d30b66078c39d625195d

dougszumski committed:
Add documentation for Monasca standalone deployment

Change-Id: I309be8e4fecb8f2f89264d84ae4a2e302f996988
Partially-Implements: blueprint monasca-roles

dougszumski committed:
Add missing project name for Monasca

Before this change the HAProxy configuration for Monasca printed:
  "Copying over haproxy haproxy config".
This change fixes the message.

TrivialFix
Change-Id: I2823775f6fa5b0be919b567f311c671f1469d19f

dougszumski committed:
Enable Kibana if Monasca enabled

Monasca uses Kibana for viewing and searching logs. A future commit
will add support for the Monasca Kibana plugin to provide tenant
scoped access to logs.

Partially-Implements: blueprint monasca-roles
Change-Id: I42b6a35f9a526ef41ff3d7cfac140fbc5596a524

dougszumski committed:
Add missing HAProxy port checks for Monasca

TrivialFix
Change-Id: Iaf216016a6acf0e9c87fdb6b8902416f4849efa3

dougszumski committed:
Automatically configure Monasca Grafana datasource

In Kolla, an OpenStack project is created to store logs and metrics
harvested from the control plane by Monasca. This commit enables
the Monasca Datasource in the Grafana organisation which maps to
this OpenStack control plane project. What this means in practice
is that if a user logs into Monasca Grafana, and has access to the
the control plane project, they will immediately be able to create
dashboards using data from Monasca which has been gathered from the
control plane.

Support to enable creation of this datasource for other OpenStack
projects can be added in a separate commit.

Partially-Implements: blueprint monasca-grafana
Change-Id: I03e741ddb1c582b7280c64637ed3e3683df6419b

dougszumski committed:
Don't set recurse on config folders

A small number of services set the recurse flag when they create
their config directory. This can change permission of files within
the directory, which are later set back to the original state. The
side effect is that the service is then restarted, even though the
net change to the config files amounts to nothing. The expected
behaviour is that a service only restarts if the config *has*
changed. This patch fixes this issue.

Change-Id: Ib6f1ca7b416247f8d455fb25892f4a3b27de03ba
Closes-Bug: 1800480

markgoddard committed:
Load configfs kernel module for iscsid

The configfs kernel module is not mounted by default in Ubuntu 16.04,
leading to the iscsid container failing to start because it bind mounts
/sys/kernel/config. The issue does not apply to Ubuntu 18.04, or other
distros (AFAIK), which load configfs by default.

This change loads the configfs module when the iscsid container is in
use.

Change-Id: I5b521ddca24b919658d2664ede2d878507d6d106
Closes-Bug: #1631072
(cherry picked from commit 5753a569c7e0ca7f352ebcaeb2386597493bf2b0)

bzurkowski committed:
Call Grafana APIs only once

In multinode deployments creating default Grafana organization failed,
because Ansible attempted to call Grafana API in the context of each
host in the inventory. After creating organization via the first host,
subsequent attempts via the remaining hosts failed due to already
existing organization. This change enforces creating default
organization only once.

Other tasks using Grafana API have been enforced to be ran only once as
well.

Change-Id: I3a93a719b3c9b4e55ab226d3b22d571d9a0f489d
Signed-off-by: Bartosz Zurkowski <b.zurkowski@samsung.com>

bzurkowski committed:
Find Monasca agent plugins locally

Find module searches paths on managed server. Since role path and custom
Kolla config is located on deployment node and deployment node is not
considered to be a managed server, Monasca plugin files cannot be found.
After the deployment container running Monasca agent collector stucks in
restart mode due to missing plugin files.

The problem does not occur if deployment was started from a managed
server (eg. OSC). The problem occurs if the deployment was started from
a separate deployment server - a common case.

This change enforces running find module locally on deployment node.

Change-Id: Ia25daafe2f82f5744646fd2eda2d255ccead814e
Signed-off-by: Bartosz Zurkowski <b.zurkowski@samsung.com>

guozijn committed:
Add customizable gnocchi volume

Change-Id: I8d21546847a2f5fa418e07cfe79016a349af4783
Signed-off-by: ZijianGuo <guozijn@gmail.com>
(cherry picked from commit 6d876934eae3f1ae2be218bccaa76f775ec68e7a)

guozijn committed:
Fix command connect to replica set primary

Without this option would cause non master error.

Change-Id: I5da653227fb1b2af7a36f39789265584ac0146da
Signed-off-by: ZijianGuo <guozijn@gmail.com>
(cherry picked from commit 30f8511cd525dadbefd232a93084e691d83df070)

Commit:
[prometheus] Support the prometheus openstack exporter

This patch implements the initial support for the
openstack-exporter[0] in the kolla-ansible
prometheus monitoring system.

The configuration and prechecks are reused from the other
exporters and a new template is provided for generating
a os-client-config file required by the exporter.

The default scrape interval is 60 seconds, but it can
be extended via a configuration option.

[0] https://github.com/Linaro/openstack-exporter

Change-Id: I4a34c4bb56e74b5cd544972cbd6540d9acb6e4a1

Conflicts:
	ansible/group_vars/all.yml
	ansible/roles/prometheus/defaults/main.yml
	ansible/roles/prometheus/templates/prometheus.yml.j2
	ansible/site.yml
	etc/kolla/globals.yml

markgoddard committed:
Fix reconfigure of neutron ml2_conf.ini

The neutron containers were not being restarted if only the ml2_conf.ini
file is changed. This is due to the XenAPI ml2_conf.ini config task
registering a variable of the same name as the task that generates
ml2_conf.ini for other services. Since the XenAPI service is typically
not running, the tasks show as not changed, and the handler skips
restarting the container.

This change adds a second variable for XenAPI to avoid this shadowing.

Change-Id: I77819ed8defb8a7653e1e5aec92013b1d40fbf02
Closes-Bug: #1783268
(cherry picked from commit 43c83b51cfcee320fc52218925e88d0fe1b05100)

Commit:
Support multi-region discovery of Nova cells

In a multi-region environment, each region is being deployed separately.
Cell discovery, however, would sometimes fail due to it picking a region
different than the one being deployed. Most likely, an internal endpoint
for region A will not be visible from region B. Furthermore, it is not
very useful to discover hosts on a region you're not modifying.

This changes the check to only run against nova compute services located
in the region being deployed.

Change-Id: I21eb1164c2f67098b81edbd5cc106472663b92cb
(cherry picked from commit 328e14253d82a7fca0a4351f6f82211435477e62)

Commit:
Hide logs when looping over passwords

When ansible goes in to a loop, by default it prints all the keys for
the item it is looping over. Some roles, when setting up the databases,
iterate over an object that includes the database password.

Override the loop label to hide everything but the database name.

Change-Id: I336a81a5ecd824ace7d40e9a35942a1c853554cd
(cherry picked from commit 30c619d1bc8938ea2177fdc21e503808dbe0cca7)

dougszumski committed:
Automate Monasca documentation for configuring Kafka

Until the Monasca Kafka client fork is removed it is currently required
to run Kafka in compatibility mode. It is also necessary to disable
an optimisation in the Kafka brokers to clean up idle connections. This
is because the optimisation was added after the Monasca Kafka client was
forked, and the client hasn't been updated since. These settings are now
applied automatically when Monasca is enabled.

Change-Id: I6935f1fb29f4f731cf3c9a70a0adf4d5812ca55e

dougszumski committed:
Make Kafka maintenance easier

This commit enables two settings which are useful when restarting a Kafka node
as part of a cluster. The first supports moving partitions for which a node
being restarting is the leader to another broker, and the second supports
automatically rebalancing partitions when the node rejoins the cluster. See
the documentation for more details:

https://kafka.apache.org/10/documentation.html#basic_ops_restarting

Change-Id: I0718fca08ea116db926b348e3e6e3bc2373db1fb

dougszumski committed:
Parse Monasca Log API timestamps correctly

By parsing the creation_time timestamp in Logstash, Elasticsearch
can parse it correctly. This closes a bug where the creation_time
timestamp was shown as a date shortly after the epoch (1970) when
viewed in Kibana.

Closes-Bug: #1816585

Change-Id: I00decfe94607845ef0eae9bec631a0e729aac3fa

dougszumski committed:
Update argument for starting Prometheus exporters

This will probably be superseded by a patch from Michal Nasiadka
but if it isn't it can go upstream as part of the Prometheus service
version upgrade.

dougszumski committed:
Default to Prometheus tag for all Prometheus images

This allows an operator to pin the Prometheus docker image
tag for all Prometheus images to that specified by the `prometheus_tag`.
Without this change, the alert manager and cadvisor tags would also need
to be set.

Change-Id: Iadef001af7d3be5b2a39ce5e2363d05a33a775e4

dougszumski committed:
Minimal changes to upgrade Elasticsearch to 6.x

dougszumski committed:
Fix Kibana registration check after E*K upgrade

This is a WIP patch which gets the ball rolling, but
currently skips configuration of the default index
which no longer works with ES 6.

See here for more info:

https://www.elastic.co/guide/en/elasticsearch/reference/6.0/removal-of-types.html

dougszumski committed:
Add mtail for exporting log events to Prometheus

This adds support for Google mtail to allow generating metrics from logs,
from which alerts may then be generated. Out of the box, metrics are
generated for user logins from syslog events. Support to add further
functionality, via the use of custom mtail programs, is supported via
Kolla configuration.

dougszumski committed:
Use prometheus_tag for Openstack exporter

Commit:
Support the prometheus elasticsearch exporter

This patch implements the support for the elasticsearch-exporter in
kolla-ansible

The configuration and prechecks are reused from the other exporters

Depends-On: Id138f12e10102a6dd2cd8d84f2cc47aa29af3972
Change-Id: Iae0eac0179089f159804490bf71f1cf2c38dde54

Conflicts:
	ansible/group_vars/all.yml
	ansible/inventory/all-in-one
	ansible/inventory/multinode
	ansible/roles/prometheus/defaults/main.yml
	ansible/roles/prometheus/handlers/main.yml
	ansible/roles/prometheus/tasks/precheck.yml
	ansible/site.yml
	etc/kolla/globals.yml
	tests/templates/inventory.j2

dougszumski committed:
Remove redundant copying of Fluentd input files

These files are copied by the wildcard which proceeds it.

Change-Id: I27dc7797db705f025c3a312cb9e9c577342ea06f

Commit:
Fix Blazar Nova aggregate in multi-region setup

The Nova aggregate was always defaulting to some region (usually first
in the Keystone endpoint list) when registering the Nova aggregate for
Blazar. Add in a region override to ensure we are always writing to the
region being deployed.

Change-Id: I3f921ac51acab1b1020a459c07c755af7023e026
(cherry picked from commit f20cbf49580f1457db22346c413991cf60c90dba)

dougszumski committed:
Fix format file message field

The message field for these format files wasn't renamed when
the input field was renamed. This problem was masked by Monasca
which renames the message field back to 'message' for the Fluentd
Monasca output plugin.

Change-Id: I2674958b1e64efddbca4765fefa55431261648e6
Closes-Bug: #1819168
(cherry picked from commit e5d564136b479e83da71ad08d31783a3f793314d)

alistarle committed:
Fix kolla-docker possible undefined variable

It is possible to reference undefined variable in kolla-docker module if
DockerWorker object initialization fail, so the current behaviour will
crash the playbook with the unwanted error message :

UnboundLocalError: local variable 'dw' referenced before assignment

Change-Id: Ic8d26b11f93255220888b5406f8ab4a6f81736c2
Closes-Bug: #1819361
(cherry picked from commit 3bb4c199817c8aa4ecd4ae684bd28f84933fb14d)

priteau committed:
Fix failure to mount cephfs

Commit 2f6b1c6890cf7ea6b0dd33ac219646e4dcaf1fd6 changed the way the
cephfs source path was generated and dropped the source path component,
keeping only the list of IPs and ports. This results in failures to
mount cephfs with the following message:

    source mount path was not specified
    failed to resolve source

Change-Id: I94d18ec064971870264ae8d0b279564f2172e548
Closes-Bug: #1819502
(cherry picked from commit 42dd7784de8d1b22cf18a7aefbc6783efe4bd7ea)

Commit:
[heat] Multi-region support for bootstrap

When bootstrapping, Heat was not setting a region explicitly, so it
could default to a region other than the one being deployed.

Change-Id: I0a0596a020fbff91ccc5b9f44f271eab220c88cd
(cherry picked from commit 44da1963ba3d089e0fc646c1b1c1fb712f1c8173)

dougszumski committed:
Add support for deploying Prometheus libvirt exporter

Add support for deploying the Kolla Prometheus libvirt exporter
image to facilitate gathering metrics from the Nova libvirt service.

Partially-Implements: blueprint libvirt-exporter

alistarle committed:
Fix action_plugins python3 compatibility

Fix filemode in the merge_configs and merge_yaml action plugin to
be compatible with python3

Change-Id: Ief64c5bdcd717141281e23c255a49ec02a96aef2
Closes-Bug: #1820134
(cherry picked from commit 85f7da979294d5bc18c6045465d2955945be3109)

dougszumski committed:
Add role to deploy cASO

cASO is an OpenStack accounting extractor. For more detail see:
https://github.com/IFCA/caso

By default, cASO is configured to output to Fluentd via TCP. The
accounting information can then be shipped off to ElasticSearch.

JohnGarbutt committed:
Add enable_ceph_rgw_external to configure haproxy

JohnGarbutt committed:
Hack to put ceph rgw in keystone catalog

dougszumski committed:
WIP: Support external Ceph for Prometheus

dougszumski committed:
Support Prometheus server 2.x

This updates command line args which have changed from the 1.x release.
Logging appears to go to stdout by default.

chenxingc committed:
Increase retries numbers of grafana check task

20 seconds may be too short to wait for grafana ready, Let's keep the
check task 60 seconds.

backport: rocky

Change-Id: Ib219ad215d1ef2147ba3591f8c398feb4f3c8888
Closes-Bug: #1821285
(cherry picked from commit 20ef211f6a045ebfdf786564b6af3936345bbdc3)

dougszumski committed:
Support adding custom alerting rules for Prometheus

Custom rules can be dropped into the prometheus/rules config folder.
Anything with the extension 'rules' will be picked up.

markgoddard committed:
Fix booting instances after nova-compute upgrade

After upgrading from Rocky to Stein, nova-compute services fail to start
new instances with the following error message:

Failed to allocate the network(s), not rescheduling.

Looking in the nova-compute logs, we also see this:

Neutron Reported failure on event
network-vif-plugged-60c05a0d-8758-44c9-81e4-754551567be5 for instance
32c493c4-d88c-4f14-98db-c7af64bf3324: NovaException: In shutdown, no new
events can be scheduled

During the upgrade process, we send nova containers a SIGHUP to cause
them to reload their object version state. Speaking to the nova team in
IRC, there is a known issue with this, caused by oslo.service performing
a full shutdown in response to a SIGHUP, which breaks nova-compute.
There is a patch [1] in review to address this.

The workaround employed here is to restart the nova compute service.

[1] https://review.openstack.org/#/c/641907

Change-Id: Ia4fcc558a3f62ced2d629d7a22d0bc1eb6b879f1
Closes-Bug: #1821362
(cherry picked from commit 192dcd1e1b9baf7f3177a694c2b1ce8bd62d9159)

markgoddard committed:
Fix up config file permissions on the host

Several config file permissions are incorrect on the host. In general,
files should be 0660, and directories and executables 0770.

Change-Id: Id276ac1864f280554e98b937f2845bb424d521de
Closes-Bug: #1821579
(cherry picked from commit a4bb8567da3e0e1468d66048c5b46cdeefcb1332)

markgoddard committed:
Check configuration file permissions in CI

Typically, non-executable files should have 660 or 600 and executable
files and directories should have 770. All should be owned by the
'config_owner_user' and 'config_owner_group' variables.

This change adds a script to check the owner and permissions of config
files under /etc/kolla, and runs it at the end of CI jobs.

Change-Id: Icdbabf36e284b9030017a0dc07b9dc81a37758ab
Related-Bug: #1821579
(cherry picked from commit 8c4ab41ffa502a58714819e0407f3c17e6fde7d0)

markgoddard committed:
Remove recurse: yes for owner/perms on /etc/kolla

When kolla-ansible bootstrap-servers is run, it executes one of the
following two tasks:

- name: Ensure node_config_directory directory exists for user kolla
  file:
    path: "{{ node_config_directory }}"
    state: directory
    recurse: true
    owner: "{{ kolla_user }}"
    group: "{{ kolla_group }}"
    mode: "0755"
  become: True
  when: create_kolla_user | bool

- name: Ensure node_config_directory directory exists
  file:
    path: "{{ node_config_directory }}"
    state: directory
    recurse: true
    mode: "0755"
  become: True
  when: not create_kolla_user | bool

On the first run, normally node_config_directory (/etc/kolla/) doesn't
exist, so it is created with kolla:kolla ownership and 0755 permissions.

If we then run 'kolla-ansible deploy', config files are created for
containers in this directory, e.g. /etc/kolla/nova-compute/. Permissions
for those files should be set according to 'config_owner_user' and
'config_owner_group'.

If at some point we again run kolla-ansible bootstrap-servers, it will
recursively set the ownership and permissions of all files in /etc/kolla
to kolla:kolla / 0755.

The solution is to change bootstrap-servers to not set the owner and
permissions recursively. It's also arguable that /etc/kolla should be
owned by 'config_owner_user' and 'config_owner_group', rather than
kolla:kolla, although that's a separate issue.

Change-Id: I24668914a9cedc94d5a6cb835648740ce9ce6e39
Closes-Bug: #1821599
(cherry picked from commit 6b0be5c5bacd91bb94ec6b75715affb9777e568d)

JohnGarbutt committed:
Fix directory for elasticsearch config

Commit:
update the event_connection to connection for panko

more infomation, refer to the panko docs[1] or code[2]

[1]: https://docs.openstack.org/panko/rocky/install/manual.html
[2]: https://github.com/openstack/panko/blob/9c4677f1784ed3353c31067e6e1bf74a0c86c037/panko/storage/__init__.py#L39

Change-Id: Ic7bdf555eb262005ca24f675cd8cee09d9bc3092
Closes-Bug: #1690843
(cherry picked from commit 47a2c5a85e9311a6b07f468b8dc7b652cc9416ba)

Commit:
Merge "Fix booting instances after nova-compute upgrade" into stable/rocky

markgoddard committed:
Don't send SIGHUP to placement-api

Seen in Kayobe CI upgrade job, placement-api exits 129 when sent a
SIGHUP. It doesn't use nova versioned objects, so doesn't need to be
reloaded.

This is a follow-up to Ia4fcc558a3f62ced2d629d7a22d0bc1eb6b879f1.

Change-Id: I38eb8645ee254cc775671c43d3c0fbd5a7402512
Related-bug: #1821362

JohnGarbutt committed:
Fix bad ceph prometheus loop

Commit:
Retry perform a synced flush task while upgrading elasticsearch

The synced flush fails due to concurrent indexing operations.
The HTTP status code in that case will be 409 CONFLICT. We can
retry this task until returns success.

Change-Id: I57f9a009b12715eed8dfcf829a71f418d2ce437b
(cherry picked from commit adb02958e7337aae3dd9d10843c2cb20fb55980b)

Commit:
add version to OPENSTACK_KEYSTONE_URL

"v3" is supposed to be part of the OPENSTACK_KEYSTONE_URL:
https://docs.openstack.org/horizon/latest/admin/customize-configure.html#configure-the-dashboard

Closes-Bug: #1822257

Change-Id: I5fd2d36305172d351fbfa9141c7cbc7c5af98f3b
(cherry picked from commit dd97c786842415c93ccc0de4788cb05ab6d012db)

jimrollenhagen committed:
Use database_address and database_port var for mariadb check

This is how services reach mariadb; verify it that way.

Closes-Bug: #1823005
Change-Id: I9924ad050118b8a853e2309654a089f65178cd77
(cherry picked from commit 524f969bfc8a96ebc2d331e241738c8ae4820b58)

Commit:
Merge "Create /volumes when using external ceph" into stable/rocky

Commit:
Merge "Fix failure to mount cephfs" into stable/rocky

Commit:
Change heat bootstrap to use internal API interface

This patch fixes Heat deployment with TLS self-signed certs.

Change-Id: Iadf67d1a5eb2b771e34d27fbced5aad15f271822
Closes-Bug: #1822990
(cherry picked from commit 86910a94db046ddb4fb0a9ef5144e3a62d0a0067)

markgoddard committed:
Make monasca notification templates optional

backport: stein

If I deploy monasca by setting enable_monasca to true, the monasca_notification
restarts with the following error:

ERROR:__main__:MissingRequiredSource: /var/lib/kolla/config_files/notification_templates/* file is not found

These templates are optional, so we need to mark this directory as optional in
config.json.

Change-Id: Ia2dd835daa7ab1153617cc92f17c2d8d498c73e0
Closes-Bug: #1823726

markgoddard committed:
Don't use easy_install on Ubuntu 18+

Currently easy_install is not available on Ubuntu 18+ due to [1]. In
that case, install pip via apt rather than easy_install.

[1]
https://bugs.launchpad.net/ubuntu/+source/python-setuptools/+bug/1774419

Change-Id: Id358fdb655d71490b0915680dff131cfe33f4a40
Closes-Bug: #1813492
(cherry picked from commit c1c02f67dcb28a8722d505ddf6a24a37c1f874d7)

markgoddard committed:
Fix bifrost tests (Rocky)

Bifrost now requires the use of the openstack CLI rather than ironic.

Change-Id: Ibb8ce5d262c03d3ddb0234cfcd41796677158f5b

Commit:
Merge "Use database_address and database_port var for mariadb check" into stable/rocky

Commit:
Merge "Add customizable gnocchi volume" into stable/rocky

Commit:
Merge "Fix command connect to replica set primary" into stable/rocky

Commit:
Merge "Fix up config file permissions on the host" into stable/rocky

Commit:
Merge "Don't use easy_install on Ubuntu 18+" into stable/rocky

Commit:
Merge "Change heat bootstrap to use internal API interface" into stable/rocky

Commit:
Merge "Check configuration file permissions in CI" into stable/rocky

dougszumski committed:
Set external web URL for Prometheus services

This change ensures that URLs returned from these services reference
the HAProxy endpoint, rather than the host on which the service is
running.

Closes-Bug: #1825150
Change-Id: I7f966ff749ea37620f1bde7019a598cb9505fa45
(cherry picked from commit 9d495504be57b30343f3f4c64f86413edfb962ad)

Conflicts:
	ansible/roles/prometheus/templates/prometheus-server.json.j2

dougszumski committed:
Set external web URL for Prometheus services

This change ensures that URLs returned from these services reference
the HAProxy endpoint, rather than the host on which the service is
running.

Closes-Bug: #1825150
Change-Id: I7f966ff749ea37620f1bde7019a598cb9505fa45
(cherry picked from commit 9d495504be57b30343f3f4c64f86413edfb962ad)

dougszumski committed:
Add support for deploying Jiralert

Jiralert is a service which runs alongside Prometheus Alertmanger to
generate Jira tickets from alerts. It provides the glue between
the Prometheus Alertmanger webhook notification service and Jira. See
https://github.com/free/jiralert for more details.

Commit:
OpenDev Migration Patch

This commit was bulk generated and pushed by the OpenDev sysadmins
as a part of the Git hosting and code review systems migration
detailed in these mailing list posts:

http://lists.openstack.org/pipermail/openstack-discuss/2019-March/003603.html
http://lists.openstack.org/pipermail/openstack-discuss/2019-April/004920.html

Attempts have been made to correct repository namespaces and
hostnames based on simple pattern matching, but it's possible some
were updated incorrectly or missed entirely. Please reach out to us
via the contact information listed at https://opendev.org/ with any
questions you may have.

markgoddard committed:
Revert "Load configfs kernel module for iscsid"

This reverts commit 4e8977c46f0cd9381020809b67061031aa83b9f2.

The module-load role is not present in the Rocky release.

Change-Id: I8fd47a83d2dab745eb2b49d4965ecb5c6b2466a4

Commit:
Merge "Revert "Load configfs kernel module for iscsid"" into stable/rocky

dougszumski committed:
Run cASO via cron

dougszumski committed:
Remove duplicate variable in groups_vars/all.yml

dougszumski committed:
Explicitly configure spool dir

This is used to store the time up to which records were last extracted.

Commit:
Check if glance_upgrading is defined before using it

glance_upgrading variable is defined only during rolling upgrades, but
its value is checked on every call to "Restart glance-api container" handler.

Closes-Bug: #1826511
Change-Id: Idc95306c50a09666ac0a12e975307a42aef9e352
(cherry picked from commit f5b52608663bea5bbe1d58eff1a7bdfd21568173)

dougszumski committed:
Add support for deploying Monasca Grafana

The Monasca Grafana fork allows users to log into Grafana with their
OpenStack user credentials and see metrics associated with their
OpenStack project. The long term goal is to enable Keystone support
in upstream Grafana, but this work seems to have stalled.

Partially-Implements: blueprint monasca-grafana
Change-Id: Icc04613b2571c094ae23b66d0bcc38b58c0ee4e1

codyhammock committed:
Add blazar to fluentd aggregation

If Blazar is enabled, ensure that fluentd processes its logs.

Change-Id: If71d5c056c042667388dae8e4ee6d51a5ecab46e
(cherry picked from commit 2c343562ee3b47248d3f0812101d48dd54793a9f)

JohnGarbutt committed:
Stop tokens being invalid due to frenet rotation

Fixes-Bug: 1809469

Right now every controller rotates frenet keys. This is nice because
should any controller die, we know the remaining ones will rotate the
keys. However, this does mean we are currently over-rotating the keys.

When we over rotate keys, we get logs like this:
This is not a recognized Fernet token <token> TokenNotFound

Most clients can recover and get a new token, but some clients (like
Nova passing tokens to other services) can't do that because it doesn't
have the password to regenerate a new token.

In crontab in keystone-fernet we see the once a day correctly staggered
across the three controllers:

ssh ctrl1 sudo cat /etc/kolla/keystone-fernet/crontab
0 0 * * * /usr/bin/fernet-rotate.sh
ssh ctrl2 sudo cat /etc/kolla/keystone-fernet/crontab
0 8 * * * /usr/bin/fernet-rotate.sh
ssh ctrl3 sudo cat /etc/kolla/keystone-fernet/crontab
0 16 * * * /usr/bin/fernet-rotate.sh

Currently we three controllers we have this in keystone.conf:

[token]
expiration = 86400 (default is 3600)
[allow_expired_window = 172800] (default is two days)
[fernet_tokens]
max_active_keys = 4

Rotation is based on pushing out a staging key, so should any server
start using that key, other servers will consider that valid. Then each
server in turn starts using the staging key, each in term demoting the
existing primary key to a secondary key. Eventually you prune the
secondary keys when there is no token in the wild that would need to be
decrypted using that key.
https://docs.openstack.org/keystone/stein/admin/fernet-token-faq.html#how-should-i-approach-key-distribution

As each controller rotates once a day, and we want to keep things valid
for three days (token expiration + allow_expired_window), plus one
staging key to promote next time and one secondary key from last time.

Change-Id: If5476e5eaf75b155bdb9965bfb475242c4deb746

JohnGarbutt committed:
Use keystone default for token expiry

Currently we default to keystone having tokens valid for one day, rather
than the one hour keystone chooses.

While it can help long running operations, like a snapshot of a very
large VM succeed, it does mean to barer of the token has access to the
cloud for a long time, even after their access has been removed in
keystone.

It might be better to default to one hour token expiry.

Change-Id: I4c86a33ca177d25425e9ec69b97144c71d0975d1

dougszumski committed:
Fix issues obtaining Keystone token with Monasca Grafana

When using the the default domain name there are issues authenticating
with Keystone. For example, you can only log in on the second attempt
and the Monasca datasource fails to authenticate. Switching to the
default domain id resolves these issues.

Change-Id: I2cb4b2608c74dd853c97e4fc27078930bc72fdf8
(cherry picked from commit fe19df124187d4f0586f3f5d365b2c7f076f3854)

markgoddard committed:
CI Test: Support being source of a Ceph upgrade

Allow Rocky release to be the source of a Ceph upgrade test.

Change-Id: Ie63a6ce6e66fb5b32e37ba2185a513aa0b05a712

markgoddard committed:
Add unit test for keystone fernet cron generator

Before making changes to this script, document its behaviour with a unit
test.

There are two major issues:

* requesting an interval of more than 1 day results in no jobs
* requesting an interval of more than 60 minutes, unless an exact
  multiple of 60 minutes, results in no jobs

Change-Id: I655da1102dfb4ca12437b7db0b79c9a61568f79e
Related-Bug: #1809469
(cherry picked from commit 25ac955a4e2645da29f8c7b807f0bac5afb43838)

markgoddard committed:
Fix keystone fernet key rotation scheduling

Right now every controller rotates fernet keys. This is nice because
should any controller die, we know the remaining ones will rotate the
keys. However, we are currently over-rotating the keys.

When we over rotate keys, we get logs like this:

 This is not a recognized Fernet token <token> TokenNotFound

Most clients can recover and get a new token, but some clients (like
Nova passing tokens to other services) can't do that because it doesn't
have the password to regenerate a new token.

With three controllers, in crontab in keystone-fernet we see the once a day
correctly staggered across the three controllers:

ssh ctrl1 sudo cat /etc/kolla/keystone-fernet/crontab
0 0 * * * /usr/bin/fernet-rotate.sh
ssh ctrl2 sudo cat /etc/kolla/keystone-fernet/crontab
0 8 * * * /usr/bin/fernet-rotate.sh
ssh ctrl3 sudo cat /etc/kolla/keystone-fernet/crontab
0 16 * * * /usr/bin/fernet-rotate.sh

Currently with three controllers we have this keystone config:

[token]
expiration = 86400 (although, keystone default is one hour)
allow_expired_window = 172800 (this is the keystone default)

[fernet_tokens]
max_active_keys = 4

Currently, kolla-ansible configures key rotation according to the following:

   rotation_interval = token_expiration / num_hosts

This means we rotate keys more quickly the more hosts we have, which doesn't
make much sense.

Keystone docs state:

   max_active_keys =
     ((token_expiration + allow_expired_window) / rotation_interval) + 2

For details see:
https://docs.openstack.org/keystone/stein/admin/fernet-token-faq.html

Rotation is based on pushing out a staging key, so should any server
start using that key, other servers will consider that valid. Then each
server in turn starts using the staging key, each in term demoting the
existing primary key to a secondary key. Eventually you prune the
secondary keys when there is no token in the wild that would need to be
decrypted using that key. So this all makes sense.

This change adds new variables for fernet_token_allow_expired_window and
fernet_key_rotation_interval, so that we can correctly calculate the
correct number of active keys. We now set the default rotation interval
so as to minimise the number of active keys to 3 - one primary, one
secondary, one buffer.

This change also fixes the fernet cron job generator, which was broken
in the following cases:

* requesting an interval of more than 1 day resulted in no jobs
* requesting an interval of more than 60 minutes, unless an exact
  multiple of 60 minutes, resulted in no jobs

It should now be possible to request any interval up to a week divided
by the number of hosts.

Change-Id: I10c82dc5f83653beb60ddb86d558c5602153341a
Closes-Bug: #1809469
(cherry picked from commit 6c1442c385450004dd253f3f464fe4336194be99)

guozijn committed:
Fix mongo command for checking replication status

By default, the mongo command connection address is '127.0.0.1:27017'.
If we change mongodb_port, It cause the handle 'Checking current
replication status' always connect to '127.0.0.1:27017'. So we need
add options like '--host' and '--port'.

Change-Id: I172ef3d19ff2bea97db3a5207c1a984cd188ceba
Signed-off-by: ZijianGuo <guozijn@gmail.com>
(cherry picked from commit 08f5c50e7491c29138049237ede070d84cc503b8)

markgoddard committed:
nova: Fix DBNotAllowed during compute startup

backport: stein, rocky

During startup of nova-compute, we see the following error message:

Error gathering result from cell 00000000-0000-0000-0000-000000000000:
DBNotAllowed: nova-compute

This issue was observed in devstack [1], and fixed [2] by removing
database configuration from the compute service.

This change takes the same approach, removing DB config from nova.conf
in the nova-compute* containers.

[1] https://bugs.launchpad.net/devstack/+bug/1812398
[2] https://opendev.org/openstack/devstack/commit/82537871376afe98a286e1ba424cf192ae60869a

Change-Id: I18c99ff4213ce456868e64eab63a4257910b9b8e
Closes-Bug: #1829705
(cherry picked from commit 002eec951f336b07ef3a7f277e9c0fe6aae9a952)

priteau committed:
Add ansible_nodename (system hostname) to /etc/hosts

Kolla-Ansible populates /etc/hosts with overcloud hosts using their API
interface IP address. When configured correctly, this allows Nova to use
the API interface for live migration of instances between compute hosts.

The hostname used is from the `ansible_hostname` variable, which is a
short hostname generated by Ansible using the first dot as a delimiter.
However, Nova defaults to use the result of socket.gethostname() to
register nova-compute services.

In deployments where hostnames are set to FQDNs, for example when using
FreeIPA, nova-compute would try to reach the other compute node using
its FQDN (as registered in the Nova database), which was absent from
/etc/hosts. This can result in failures to live migrate instances if
DNS entries don't match.

This commit populates /etc/hosts with `ansible_nodename` (hostname as
reported by the system) in addition to `ansible_hostname`, if they are
different.

Change-Id: Id058aa1db8d60c979680e6a41f7f3e1c39f98235
Closes-Bug: #1830023
(cherry picked from commit 37899026bf7ec6ae70b01571cd0fb2e0173f8f9b)

dougszumski committed:
Set log message field name in Monasca Fluentd plugin

Let the Monasca Fluentd output plugin handle mapping of the log
message.

Change-Id: I4a74a91b9b38d5c172397a7e7204e626bcedcfac
Closes-Bug: #1830184
Depends-On: https://review.opendev.org/#/c/660988/
(cherry picked from commit a5cd622396700ff995d5fa7f70ff5bdeb2156013)

dougszumski committed:
Remove obsolete roles middleware

Change I0ca38f2cc7d63b9b47eedb304ba7b00a94816f9a removed the roles
middleware from the example paste pipeline.

Change-Id: Ie9a3b0fef395aaf414407f6bae1ac4bca158240d
(cherry picked from commit f23901677cdd247315eea56150df1fafafb107d1)

Commit:
Pin ARA to <1.0.0 and use stable setup interface to find callback plugin

ARA 1.0 will be released in the near future and isn't backwards
compatible.
Pin it so it doesn't break things unexpectedly.

ARA ships simple setup modules to help figure out the paths
to where things are located.
These are backwards compatible from ARA 1.0 to 0.x.

Change-Id: Ie4dbaf54b2fee1ab9a6d5593fe34267dcd4e91db
(cherry picked from commit 86e9b0831c89352d062bb828e28af33240b94f89)

guozijn committed:
Update service_provider for FWaaS v2

As described in document [1], the service_provider should be set to
'FIREWALL_V2:fwaas_db:neutron_fwaas.services.firewall.service_drivers.
agents.agents.FirewallAgentDriver:default' for FWaaS v2.

[1] https://docs.openstack.org/neutron/stein/admin/fwaas-v2-scenario.html#enable-fwaas-v2

Change-Id: Iad3e906af325ba3209a77a91ab2061e515b4a219
Signed-off-by: ZijianGuo <guozijn@gmail.com>
(cherry picked from commit b374ea33e4c90bb393c062e97737b90fd2bc455c)

Commit:
Elevate privileges for copying gnocchi policy.json

Change-Id: I616a30acf38a61265f3005c712767d5221abf5c5
Closes-Bug: #1830723
(cherry picked from commit 4a60dc21e5f845d834247caf423801d84b5c987a)

dincercelik committed:
Fixes VMTP deploys.

Merged config files outputs to python_path which is incorrect. Changing it
to node_config_directory.

Change-Id: If801f54709114b931677adb605dffb75cfab25cd
Closes-Bug: #1830346
(cherry picked from commit 56b2e947a1afae30ef21113717f61e96116aa873)

Wasaac committed:
Fix monasca grafana organisation check

"Create default control plane organisation if it doesn't exist" task
fails when organisation already exists.
The list organisation task currently returns project domain id.
The create organisation task currently provides project domain name.
Change the create task to use default_project_domain_id instead.

TrivialFix

Change-Id: Ice70d55e6729fe55164dcf85e98acdc1d7925209
(cherry picked from commit 84edfd09b67ef28e40b3ff24b96afa0ded5c532e)

Commit:
Merge "Constrain the size of Docker logs" into stable/rocky

Commit:
Merge "Increase retries numbers of grafana check task" into stable/rocky

Commit:
Merge "Use alembic migrations to bootstrap Monasca database" into stable/rocky

Commit:
Merge "Remove Kafka partitions config option" into stable/rocky

Commit:
Merge "Support configuring Monasca log pipeline performance" into stable/rocky

Commit:
Merge "Support deploying Monasca Log Metrics" into stable/rocky

Commit:
Merge "Improve scalability of Kafka" into stable/rocky

Commit:
Merge "Support deploying an Apache Storm cluster" into stable/rocky

Commit:
Merge "Support deploying Monasca Thresh" into stable/rocky

Commit:
Merge "Support deploying Monasca Notification engine" into stable/rocky

Commit:
Merge "Add some missing parameters for Monasca Notification" into stable/rocky

Commit:
Merge "Support deploying Monasca Persister" into stable/rocky

Commit:
Merge "Add missing Monasca persister and notification dimensions" into stable/rocky

Commit:
Merge "Set external web URL for Prometheus services" into stable/rocky

Commit:
Merge "Fix format file message field" into stable/rocky

Commit:
Merge "Support configuring Monasca Persister performance" into stable/rocky

Commit:
Merge "Add region to Monasca Log API" into stable/rocky

Commit:
Merge "Improve registration of Monasca InfluxDB database" into stable/rocky

Commit:
Merge "Disable InfluxDB reporting" into stable/rocky

Commit:
Merge "Support custom monasca-notification templates" into stable/rocky

Commit:
Merge "Prevent Monasca Notification using YAML mode" into stable/rocky

Commit:
Merge "Increase maximum payload size for Monasca Log API" into stable/rocky

Commit:
Merge "Update Monasca Log Persister config" into stable/rocky

Commit:
Merge "Update Monasca Log Transformer config" into stable/rocky

Commit:
Merge "Support using binary Logstash image" into stable/rocky

Commit:
Merge "Forward system logs to Monasca Log API" into stable/rocky

Commit:
Merge "Don't set recurse on config folders" into stable/rocky

Commit:
Merge "Add initial documentation for Monasca" into stable/rocky

Commit:
Merge "Support deploying the Monasca Agent" into stable/rocky

Commit:
Merge "Monasca specific config for Fluentd" into stable/rocky

Commit:
Merge "Add missing project name for Monasca" into stable/rocky

Commit:
Merge "Add missing HAProxy port checks for Monasca" into stable/rocky

Commit:
Merge "Add support for deploying Monasca Grafana" into stable/rocky

Commit:
Merge "Add documentation for Monasca standalone deployment" into stable/rocky

Commit:
Merge "Automatically configure Monasca Grafana datasource" into stable/rocky

Commit:
Merge "Automate Monasca documentation for configuring Kafka" into stable/rocky

Commit:
Merge "Make Kafka maintenance easier" into stable/rocky

Commit:
Merge "Parse Monasca Log API timestamps correctly" into stable/rocky

Commit:
Merge "Make monasca notification templates optional" into stable/rocky

Commit:
Merge "Update log metrics config for Logstash downgrade" into stable/rocky

Commit:
Merge "Call Grafana APIs only once" into stable/rocky

Commit:
Merge "Enable Kibana if Monasca enabled" into stable/rocky

Commit:
Merge "Find Monasca agent plugins locally" into stable/rocky

Commit:
Merge "Fix issues obtaining Keystone token with Monasca Grafana" into stable/rocky

Commit:
Merge "Fix bifrost tests (Rocky)" into stable/rocky

priteau committed:
Stop duplicating Nova cells

Check if a base Nova cell already exists before calling `nova-manage
cell_v2 create_cell`, which would otherwise create a duplicate cell when
the transport URL or database connection change.

If a base cell already exists but the connection values have changed, we
now call `nova-manage cell_v2 update_cell` instead. This is only
possible if a duplicate cell has not yet been created. If one already
exists, we print a warning inviting the operator to perform a manual
cleanup. We don't use a hard fail to avoid an abrupt change of behavior
if this is backported to stable branches.

Change-Id: I7841ce0cff08e315fd7761d84e1e681b1a00d43e
Closes-Bug: #1734872
(cherry picked from commit 19b8dbe460315c5da23c33aa57c967d75d7f2018)

Commit:
Merge "Set log message field name in Monasca Fluentd plugin" into stable/rocky

Commit:
Merge "CI Test: Support being source of a Ceph upgrade" into stable/rocky

dougszumski committed:
Fix issue finding custom, host specific plugins

The results from the find operation need to be registered per host,
because they depend on the host which runs the search. This bug
impacts users specifying custom plugins for specific hosts.

Change-Id: I41b2986b2f4ccd8fdc6553e83737e4106b6a2c07
(cherry picked from commit 4be219fcb7f3ed188ea0c4c5b2cb9a1fdba51991)

angeiv committed:
Fix mongo command for supporting other port

If we change mongodb_port, the command for bootstraping mongodb
should not connect to the default mongodb port 27017.

Change-Id: I330999be577d6416df162ea33fa1f7a19df56029
(cherry picked from commit f7bfce20ada638b80bf06d0a8134ca9e01dea273)

yoctozepto committed:
Avoid parallel discover_hosts (nova-related race condition)

In a rare event both kolla-ansible and nova-scheduler try to do
the mapping at the same time and one of them fails.
Since kolla-ansible runs host discovery on each deployment,
there is no need to change the default of no periodic host discovery.

I added some notes for future. They are not critical.
I made the decision explicit in the comments.
I changed the task name to satisfy recommendations.
I removed the variable because it is not used (to avoid future doubts).

Closes-Bug: #1832987
Change-Id: I3128472f028a2dbd7ace02abc179a9629ad74ceb
Signed-off-by: Radosław Piliszek <radoslaw.piliszek@gmail.com>
(cherry picked from commit ce680bcfe2208f19a23630ee3feff4b9bd40c595)

Commit:
Merge "Configure coordination in default for cinder when redis is enabled" into stable/rocky

Commit:
Merge "Stop duplicating Nova cells" into stable/rocky

Commit:
Merge "Elevate privileges for copying gnocchi policy.json" into stable/rocky

Commit:
Merge "Remove obsolete roles middleware" into stable/rocky

Commit:
Merge "Fix issue finding custom, host specific plugins" into stable/rocky

Commit:
Merge "Support multi-region discovery of Nova cells" into stable/rocky

Commit:
Merge "Add blazar to fluentd aggregation" into stable/rocky

Commit:
Merge "Fix Blazar Nova aggregate in multi-region setup" into stable/rocky

Commit:
Merge "Hide logs when looping over passwords" into stable/rocky

Commit:
Merge "Fix monasca grafana organisation check" into stable/rocky

Commit:
Merge "Fixes VMTP deploys." into stable/rocky

Commit:
Merge "Add ansible_nodename (system hostname) to /etc/hosts" into stable/rocky

Commit:
Merge "[heat] Multi-region support for bootstrap" into stable/rocky

Commit:
Merge "Fix mongo command for supporting other port" into stable/rocky

Commit:
Merge "Update service_provider for FWaaS v2" into stable/rocky

JohnGarbutt committed:
Remove log arg that fails with prometheus 2.0

dougszumski commented on ansible/roles/prometheus/templates/prometheus-server.json.j2:
```diff
@@ -1,5 +1,5 @@
 {
-    "command": "/opt/prometheus/prometheus --config.file /etc/prometheus/prometheus.yml --web.listen-address {{ api_interface_address }}:{{ prometheus_port }} --web.external-url={{ internal_protocol }}://{{ kolla_internal_fqdn }}:{{ prometheus_port }} --log.format logger:stdout --storage.tsdb.path /var/lib/prometheus",
+    "command": "/opt/prometheus/prometheus --config.file /etc/prometheus/prometheus.yml --web.listen-address {{ api_interface_address }}:{{ prometheus_port }} --web.external-url={{ internal_protocol }}://{{ kolla_internal_fqdn }}:{{ prometheus_port }} --storage.tsdb.path /var/lib/prometheus",
```
Ah yes, I need to push the 2.0 upgrade upstream..

Commit:
Merge "Add unit test for keystone fernet cron generator" into stable/rocky

Commit:
Merge "Fix keystone fernet key rotation scheduling" into stable/rocky

yoctozepto committed:
Remove zuul-cloner usage

Due to zuul-cloner being finally removed from Zuul v3, we have to
remove zuul-cloner usage from supported branches to use CI.
Stein and up have it removed by an unrelated change.
This simple patch is to go to rocky and below.

See [1] for Zuul change.

[1] https://review.opendev.org/663151

Change-Id: I4b98c2cf207e3511e00ae69e88a9b5b126daa6c9
Signed-off-by: Radosław Piliszek <radoslaw.piliszek@gmail.com>

jovial committed:
Specify endpoint when creating monasca user

otherwise I'm seeing:

TASK [monasca : Creating the monasca agent user] ****************************************************************************************************************************
fatal: [monitor1]: FAILED! => {"changed": false, "module_stderr": "Shared connection to 172.16.3.24 closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  F
ile \"/tmp/ansible_I0RmxQ/ansible_module_kolla_toolbox.py\", line 163, in <module>\r\n    main()\r\n  File \"/tmp/ansible_I0RmxQ/ansible_module_kolla_toolbox.py\", line 141,
 in main\r\n    output = client.exec_start(job)\r\n  File \"/opt/kayobe/venvs/kolla-ansible/lib/python2.7/site-packages/docker/utils/decorators.py\", line 19, in wrapped\r\n
    return f(self, resource_id, *args, **kwargs)\r\n  File \"/opt/kayobe/venvs/kolla-ansible/lib/python2.7/site-packages/docker/api/exec_api.py\", line 165, in exec_start\r\
n    return self._read_from_socket(res, stream, tty)\r\n  File \"/opt/kayobe/venvs/kolla-ansible/lib/python2.7/site-packages/docker/api/client.py\", line 377, in _read_from_
socket\r\n    return six.binary_type().join(gen)\r\n  File \"/opt/kayobe/venvs/kolla-ansible/lib/python2.7/site-packages/docker/utils/socket.py\", line 75, in frames_iter\r\
n    n = next_frame_size(socket)\r\n  File \"/opt/kayobe/venvs/kolla-ansible/lib/python2.7/site-packages/docker/utils/socket.py\", line 62, in next_frame_size\r\n    data = read_exactly(socket, 8)\r\n  File \"/opt/kayobe/venvs/kolla-ansible/lib/python2.7/site-packages/docker/utils/socket.py\", line 47, in read_exactly\r\n    next_data = read(socket, n - len(data))\r\n  File \"/opt/kayobe/venvs/kolla-ansible/lib/python2.7/site-packages/docker/utils/socket.py\", line 31, in read\r\n    return socket.recv(n)\r\nsocket.timeout: timed out\r\n", "msg": "MODULE FAILURE", "rc": 1}

when the monitoring nodes aren't on the public API network.

Change-Id: I7a93f69da0e02c9264da0b081d2e60626f899e3a
(cherry picked from commit 9074da56a79adcfd0f9348f703a15ac3ebd486ab)

Commit:
Merge "Avoid parallel discover_hosts (nova-related race condition)" into stable/rocky

yoctozepto committed:
(Rocky and Queens only) CI: Fix ceph jobs for kolla

The way Zuul handles job dependencies caused the computed path
to setup_ceph_disks.sh to be wrong when job was run due to kolla
(as opposed to kolla-ansible) change.

This patch uses a relative path as is the case in Stein and later.

Change-Id: I4f45cd18b86ebe0efcc3101e31e4bcafdc83eb57
Signed-off-by: Radosław Piliszek <radoslaw.piliszek@gmail.com>

yoctozepto committed:
CI: set the same gate queue for kolla and kolla-ansible

This is to ensure that any Depends-On does not cause Zuul not to pick up
the change for gating due to no notifications between queues.
Previously W+1-ing a change which depended on non-merged change from
the other project caused it to remain in the same state.

Change-Id: Ib2d88471ac5730c00b5a9721066d1fb3f2998c9c
Signed-off-by: Radosław Piliszek <radoslaw.piliszek@gmail.com>
(cherry picked from commit 9c815fa50350bae7a2c9b1edb17611707f9506f0)

paulbrowne committed:
Correct typo in this Boolean

JohnGarbutt committed:
Merge pull request #25 from RSE-Cambridge/prom_jiralert_typo

Correct typo in this Boolean

JohnGarbutt committed:
Merge remote-tracking branch 'origin/cumulus/rocky' into upstream-sync-june6

Change-Id: I08dc82aa4aa805aca8d45e2a8b9a65b7eefc2ed5

JohnGarbutt committed:
Merge remote-tracking branch 'upstream/stable/rocky' into upstream-sync-june6

Conflicts:
	ansible/roles/keystone/templates/keystone.conf.j2

Change-Id: I15cf1f4f20a682b284dfa2611320525ada1a47d5

JohnGarbutt commented:
I have actually pulled in more commits from stable/rocky, mostly to pull in the updated keystone tokens fix
